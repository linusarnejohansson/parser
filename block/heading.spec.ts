import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("block.heading", () => {
	it("level 1", () => {
		const result = parse(new Source(IO.StringReader.create("# Heading\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("level 2", () => {
		const result = parse(new Source(IO.StringReader.create("## Heading\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("level 3", () => {
		const result = parse(new Source(IO.StringReader.create("### Heading\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("level 4", () => {
		const result = parse(new Source(IO.StringReader.create("#### Heading\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("level 5", () => {
		const result = parse(new Source(IO.StringReader.create("##### Heading\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("level 6", () => {
		const result = parse(new Source(IO.StringReader.create("###### Heading\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
