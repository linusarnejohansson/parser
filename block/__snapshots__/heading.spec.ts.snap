// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.heading level 1 1`] = `
Array [
  Object {
    "class": "Block.Heading",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Heading",
      },
    ],
    "level": 1,
  },
]
`;

exports[`block.heading level 2 1`] = `
Array [
  Object {
    "class": "Block.Heading",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Heading",
      },
    ],
    "level": 2,
  },
]
`;

exports[`block.heading level 3 1`] = `
Array [
  Object {
    "class": "Block.Heading",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Heading",
      },
    ],
    "level": 3,
  },
]
`;

exports[`block.heading level 4 1`] = `
Array [
  Object {
    "class": "Block.Heading",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Heading",
      },
    ],
    "level": 4,
  },
]
`;

exports[`block.heading level 5 1`] = `
Array [
  Object {
    "class": "Block.Heading",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Heading",
      },
    ],
    "level": 5,
  },
]
`;

exports[`block.heading level 6 1`] = `
Array [
  Object {
    "class": "Block.Heading",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Heading",
      },
    ],
    "level": 6,
  },
]
`;
